api/kv_v2.go

```diff
@@ -238,6 +258,41 @@ func (kv *kvv2) Put(ctx context.Context, secretPath string, data map[string]inte
 	return kvSecret, nil
 }
 
+// PutMetadata can be used to fully replace a subset of metadata fields for a
+// given KV v2 secret.
```

Comment:
The confusing part about the KVv2 metadata API is that it _mostly_ acts as a patch operation today. In other words, it acts as a patch for all the top-level scalar values. It does _not_ provide patch support for the fields nested under the `custom_metadata` field.

Reply:
Interesting... my tests seem to show PatchMetadata working as expected on the `custom_metadata` field, at least with regard to the checks I've got [here](https://github.com/hashicorp/vault/blob/1773d0b153802b0478358e02f1a89f883ce100ae/vault/external_tests/api/kv_helpers_test.go#L521-L536). 

Or did you just mean that regular PutMetadata will perform a full overwrite of the contents of the custom_metadata field? If so, I think that's fair to expect from a PUT.